### PR TITLE
Log more info when raising errRootFSMismatch

### PR DESCRIPTION
**Pull request by lmbarros**
We currently do not log any information to help us understanding the
underlying issue -- not even to identify what is the exact point in
which the error is raised. This commit improves on this situation.

Meant to help with #244.

### CODE REVIEW

File: distribution/pull_v2.go [hunk 1]
```diff
@@ -665,6 +665,11 @@ func (p *v2Puller) pullSchema2Layers(ctx context.Context, target distribution.De
 	}
 
 	if len(descriptors) != len(configRootFS.DiffIDs) {
+		logrus.Errorf("RootFSMismatch: len(descriptors) = %v, len(configRootFS.DiffIDs) = %v",
+			len(descriptors), len(configRootFS.DiffIDs))
+		logrus.Errorf("descriptors = %#v", descriptors)
+		logrus.Errorf("configRootFS = %#v", configRootFS)
```
Comment (robertgzr) [inline distribution/pull_v2.go:671]: how about making this a single line? printing those values like that will probably make them very hard to parse like that, but it keeps all the information together, which should be useful when looking at the full journal output of a busy device

```suggestion
		logrus.WithFields(logrus.Fields{
			"len(descriptors)": len(descriptors),
			"len(configRootFS.DiffIDs)": len(configRootFS.DiffIDs),
			"descriptors": fmt.Sprintf("%#v", descriptors),
			"configRootFS": fmt.Sprintf("%#v", configRootFS),
		}).Error("rootFS mismatch")
```

Comment (lmbarros) [reply distribution/pull_v2.go]: Right, good point, thanks. I updated the branch!

**Comment by lmbarros** [comment]
@balena-ci rebase

**Comment by lmbarros** [comment]
@balena-ci I self-certify!
